modules/storages/app/controllers/storages/admin/storages/project_storages_controller.rb

```diff
@@ -73,10 +72,24 @@ def create
     respond_with_turbo_streams(status: create_service.success? ? :ok : :unprocessable_entity)
   end
 
-  def destroy; end
+  def destroy_confirmation_dialog
+    respond_with_dialog Storages::ProjectStorages::DestroyConfirmationDialogComponent.new(
+      storage: @storage,
+      project_storage: @project_storage
+    )
+  end
+
+  def destroy
+    @project_storage.destroy
```

Comment:
🟠 There's an edge case where a user can attempt to unlink a project from different tabs- in which case they would be met by an unhandled 404- in project attributes, we handled this via a before action [here](https://github.com/opf/openproject/blob/dev/app/controllers/admin/settings/project_custom_fields_controller.rb#L186-L189) shall we extend the same?

----


https://github.com/user-attachments/assets/37fbee7f-5817-409e-a5cd-e391ad6184c4

Reply:
See https://github.com/opf/openproject/pull/15653

Reply:
Thanks for pointing that out!